docs/caa.md

```diff
@@ -0,0 +1,107 @@
+Authorities (CAs) are allowed to issue certificates containing their domain names. It
+was standardized in 2013 by [RFC 6844](https://tools.ietf.org/html/rfc6844) to
+allow a CA "reduce the risk of unintended certificate mis-issue." By default,
+every public CA is allowed to issue certificates for any domain name in the
+public DNS, provided they validate control of that domain name. That means that
+if there's a bug in any one of the many public CAs' validation processes, every
+domain name is potentially affected. CAA provides a way for domain holders to
+reduce that risk.
+
+# Using CAA
+
+If you don't care about CAA, you generally don't have to do anything (but see
+CAA errors below). If you would like to use CAA to restrict which Certificate
+Authorities are allowed to issue certificates for your domain, you will need to
+use a DNS provider that supports setting CAA records. Check [SSLMate's CAA
+page](https://sslmate.com/labs/caa/#support) for a list of such providers. If your
+provider is listed, you can use
+[SSLMate's CAA Record Generator](https://sslmate.com/labs/caa/) to generate a
+set of CAA records listing the CAs that you would like to allow.
+
+## Where to put the record
+
+You can set CAA records on your main domain, or at any depth of subdomain.
+For instance, if you had `www.community.example.com`, you could set CAA records
+for the full name, or for `community.example.com`, or for `example.com`. CAs
+will check each version, from left to right, and stop as soon as they see any
+CAA record. So for instance, a CAA record at `community.example.com` would take
+precedence over one at `example.com`. Most people who add CAA records will want
+to add them to their registered domain (`example.com`) so that they apply to all
+subdomains.
+
+CAA validation follows CNAMEs, like all other DNS requests. If
+`www.community.example.com` is a CNAME to `web1.example.net`, the CA will first
+request CAA records for `www.community.example.com`, then seeing that there is a
+CNAME for that domain name instead of CAA records, will request CAA records for
+`web1.example.net` instead. Note that if a domain name has a CNAME record, it is
+not allowed to have any other records according to the DNS standards.
+
+The [CAA RFC](https://tools.ietf.org/html/rfc6844) specifies an additional
+behavior called "tree-climbing" that requires CAs to also check the parent
+domains of the result of CNAME resolution. Let's Encrypt does not implement tree
+climbing because it makes expressing certain CAA policies impossible. After
+[discussion on the IETF mailing
+list](https://mailarchive.ietf.org/arch/msg/spasm/vHL260X6Zb2C0VSwDwa8VZC1Kw0),
+we achieved consensus that tree-climbing in CAA is not ideal, and there's [an
+erratum](https://www.rfc-editor.org/errata/eid5065) for the CAA RFC removing it.
+That erratum still needs to be voted in by the CA/Browser Forum by September 8
+for it to be official.
+
+# CAA errors
```

Comment:
This looks great: I think one possible omission is the discussion of round-robin or anycast authoritative DNS. I think it would head off some "But it works for me!" confusion if we mention that often what _seems_ like one authoritative DNS server can be many behind the scenes and which is used can be tied to the source network of your queries.